src/Chainweb/Pact/Backend/ChainwebPactDb.hs

```diff
@@ -440,15 +441,22 @@ doCreateUserTable tn@(TableName ttxt) mn = do
     txlogs = DL.singleton (TxLog txlogKey stn (toJSON uti))
 {-# INLINE doCreateUserTable #-}
 
+logit :: Show a => a -> BlockHandler SQLiteEnv ()
+logit a = asks _logger >>= \l -> liftIO $ logLog l "ERROR" ("logit: " ++ show a)
+
 doRollback :: BlockHandler SQLiteEnv ()
-doRollback = bsPendingTx .= Nothing
+doRollback = do
+  logit ("TX","doRollback")
+  resetTemp
```

Comment:
I think this will break the checkpointer. If you've had successful txs played in the block, but you play a subsequent tx, and roll it back, won't the `resetTemp` clear out all of the pending transactions for the block? We _do_ need to set `bsMode .= Nothing` here though.

Reply:
ok. I confess to not understanding how the txlog works in the pending stuff ...